core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -197,7 +204,8 @@ class ReplicaManager(val config: KafkaConfig,
                      val delayedDeleteRecordsPurgatory: DelayedOperationPurgatory[DelayedDeleteRecords],
                      val delayedElectLeaderPurgatory: DelayedOperationPurgatory[DelayedElectLeader],
                      threadNamePrefix: Option[String],
-                     val alterIsrManager: AlterIsrManager) extends Logging with KafkaMetricsGroup {
+                     val alterIsrManager: AlterIsrManager,
+                     usingRaftMetadataQuorum: Boolean) extends Logging with KafkaMetricsGroup {
```

Comment:
Could we check if we're in Raft mode using KafkaConfig rather than adding a flag here?